File: Python/Recursividad1.py
class Nodo:
    def __init__(self,numero):
        self.num = numero
        self.prox = None 

class Lista:
    def __init__(self):
        self.prim = None

    def cargarNodo(self,numero):
        nuevo = Nodo(numero) 
        nuevo.prox = self.prim
        self.prim = nuevo 

    def mostrarNodo(self):
        p = self.prim
        while p is not None:
            print(f"Número: {p.num}")
            p = p.prox


def esPar(n: int) -> bool:
    if (n % 2) == 0:
        return True 
    else: 
        return False 

lista = Lista()

arreglo = [
    1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
    11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
    21, 22, 23, 24, 25, 26, 27, 28, 29, 30
]

def verificar(a,indice):
    if indice < 30:
        if esPar(a[indice]):
            lista.cargarNodo(a[indice])

        return verificar(a,indice + 1)


verificar(arreglo,0)

lista.mostrarNodo()

File: Random/Recursividad1.go
"7. A partir de un arreglo de entrada de 100 posiciones de enteros, se pide cargar en
una lista los elementos del arreglo que son impares. Para la verificación de paridad
se pide desarrollar una función recursiva."

Accion recursiva (numeros: arreglo[1...10] de enteros) es 
 Ambiente
	nodo = registro 
		num: entero 
		prox: puntero a nodo 
	FReg
	prim,q,p: puntero a nodo 

	Procedimiento cargarNodo(numero)
		Nuevo(q); *q.num:= numero
		*q.prox:= prim 
		prim:= q
	FProcedimiento

	Funcion esPar(n:entero):booleano
		si n mod 2 = 0 entonces 
			esPar:= verdadero
		sino 
			esPar:= falso 
		Fsi
	FFuncion

	Procedimiento verificar(a: arreglo[1...100] de enteros; indice: entero)
		si ind < 101 entonces 
			si esPar(a[indice]) entonces 
				cargarNodo(a[indice])
			Fsi
		Fsi

		verificar(a,indice + 1)
	FProcedimiento

	Proceso 
		prim:= nil 
		verificar(numeros,0)

		p:= prim 
		Mientras p <> nil hacer 
			Esc(*p.num)
		FM 
	FProceso 
FAccion

